compiler/hash-typecheck/src/ops/oracle.rs

```diff
@@ -56,6 +56,20 @@ impl<'tc> Oracle<'tc> {
         }
     }
 
+    /// Check if the [Term] is a primitive type.
+    pub fn term_is_primitive(&self, term: TermId) -> bool {
```

Comment:
You're mixing literal values and primitive types here

Reply:
Yes, this is to find out if it is of a **primitive** type, ideally, this would handle values as well, and just try to get the type of it.